File: 2023/a.py
#!/usr/bin/env python3

# Problem A - Which Team Should Receive the Sponsor Prize?

def solve(a):
    _, i = min((abs(ai - 2023), i) for i, ai in enumerate(a))
    return i + 1

while True:
    n = int(input())
    if n == 0:
        break
    a = list(map(int, input().split()))
    assert len(a) == n
    print(solve(a))

File: 2023/elixir/a.ex
#!/usr/bin/env elixir

# ACM ICPC 2023 domestic qualifier
# A - Which Team Should Receive the Sponsor Prize?

defmodule Main do
  def main do
    n = IO.gets("") |> String.trim |> String.to_integer
    if n != 0 do
      a = IO.gets("") |> String.split |> Enum.map(&String.to_integer/1)
      solve(n, a) |> IO.inspect
      main()
    end
  end
  
  def solve(n, a) do
    a |> Enum.map(&(abs(&1 - 2023))) |> Enum.zip(1..n)
      |> Enum.min |> elem(1)
  end
end

Main.main
